Tests/ExtData_Testing_Framekwork/CMakeLists.txt

```
@@ -0,0 +1,57 @@
+# Detect if we are using Open MPI and add oversubscribe
+string(REPLACE " " ";" MPI_Fortran_LIBRARY_VERSION_LIST ${MPI_Fortran_LIBRARY_VERSION_STRING})
+list(GET MPI_Fortran_LIBRARY_VERSION_LIST 0 MPI_Fortran_LIBRARY_VERSION_FIRSTWORD)
+if(MPI_Fortran_LIBRARY_VERSION_FIRSTWORD MATCHES "Open")
+   list(APPEND MPIEXEC_PREFLAGS "-oversubscribe")
+endif()
+
+file(STRINGS "test_cases/cases.txt" TEST_CASES)
+
+option(TEST_EXTDATA1G "Test ExtData 1G" ON)
+option(TEST_EXTDATA2G "Test ExtData 2G" ON)
+
+set(cutoff "7")
+
+foreach(TEST_CASE ${TEST_CASES})
+  if (EXISTS ${CMAKE_CURRENT_LIST_DIR}/test_cases/${TEST_CASE}/nproc.rc)
+     file(READ ${CMAKE_CURRENT_LIST_DIR}/test_cases/${TEST_CASE}/nproc.rc num_procs)
+  else()
+     set(num_procs "1")
+  endif()
+  if (${TEST_EXTDATA1G})
+     add_test(
+        NAME "ExtData1G_${TEST_CASE}"
+        COMMAND ${CMAKE_COMMAND} 
+          -DTEST_CASE=${TEST_CASE} 
+          -DMPIEXEC_EXECUTABLE=${MPIEXEC_EXECUTABLE} 
+          -DMPIEXEC_NUMPROC_FLAG=${MPIEXEC_NUMPROC_FLAG}
+          -DMY_BINARY_DIR=${CMAKE_BINARY_DIR}/bin
```

```suggestion
          -DMY_BINARY_DIR=${CMAKE_CURRENT_BINARY_DIR}/bin
```
BINARY is top of tree, CURRENT_BINARY is relative to source.   When the test is built inside full geos this makes a big difference.

Not sure `bin` is the best name for the subdir.

---

I'll try this in the model.

---

I tested this in a full model build. The tests worked so what Matt and I ended up doing here appears to be correct. We didn't choose bin, that's the binary directory in the build so no choice there...